rfc/rfc022-discovery-service.md

```diff
@@ -0,0 +1,252 @@
+
+For example:
+
+``json
+{
+  "id": "uc_university_v1",
+  "endpoint": "https://example.com/usecase/university/v1",
+  "presentation_max_validity": 259200,
+  "presentation_definition": {
+    "id": "pd_university",
+    "input_descriptors": [
+      {
+        "id": "pd_university_type",
+        "constraints": {
+          "fields": [
+            {
+              "path": ["$.type"],
+              "filter": {
+                "type": "string",
+                "const": "UniversityCredential"
+              }
+            },
+            {
+              "path": "$.credentialSubject.name",
+              "filter": {
+                "type": "string"
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+``
+
+## 6. Trust
+
+Trust of credential issuers (e.g. `did:example:education-accredetor` issuing `EducationalInstitutionCredential`) SHOULD be defined by the presentation definition.
+In this case, there should be 2 constraints: one for the type and one for the issuer:
+
+``json
+[
+  {
+    "path": ["$.type"],
+    "filter": {
+      "type": "string",
+      "const": "EducationalInstitutionCredential"
+    }
+  },
+  {
+    "path": "$.issuer",
+    "filter": {
+      "type": "string",
+      "const": "did:example:education-accredetor"
+    }
+  }
+]
+``
+
+## Appendix A - Design Decisions
+
+### Supporting multiple lists
+
+This RFC does not specify nested or multiple lists as this can be achieved by hosting a separate list on an alternate HTTP endpoint.
+
+### Using deltas to reduce indexing overhead
+
+When only the full list is available, clients need to validate and re-index (for fulltext search) all presentations.
+By having the server return only new presentations instead, clients can only process those new entries.
+
+### Tag implementation
+
+The tag is specified as an opaque value (JSON string). It is up to server to decide how to implement this.
+
+One way to implement this is to use a [Lamport timestamp](https://en.wikipedia.org/wiki/Lamport_timestamp) which is incremented for every presentation received.
+That way, timestamps unambiguously reference the last presentation the client received.
+
+A Unix timestamp does not offer this property, as it is possible that multiple presentations are received at the second,
+possibly serving clients duplicate presentations.
+
+## References
+
+* [JWT] M. Jones, J. Bradley, N. Sakimura, "JSON Web Token (JWT)", RFC 7519, May 2015, <https://www.rfc-editor.org/info/rfc7519>.
+* [VC-DATA-MODEL] M. Sporny, D. Longley, D. Chadwick, "Verifiable Credentials Data Model 1.1", W3C Recommendation, 3 March 2022, <https://www.w3.org/TR/vc-data-model/>.
+* [PE] D. Buchner, B. Zundel, M. Riedel, K.H. Duffy, "Presentation Exchange 2.0.0", 12 September 2023, <https://identity.foundation/presentation-exchange/spec/v2.0.0/>.
```

Comment:
This is usually referred to as `PEX` since that sounds a lot nicer when pronounced.